### PR TITLE
JENKINS-28629 - Fix Apache Maven 3.3 support - create maven33 agent a…

### PR DESCRIPTION
**Pull request by tomerc**
…nd interceptor

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -60,6 +62,7 @@ THE SOFTWARE.
     <maven3.0.x.version>3.0.5</maven3.0.x.version>
     <maven3.1.x.version>3.1.0</maven3.1.x.version>
     <maven3.2.x.version>3.2.3</maven3.2.x.version>
+    <maven3.3.x.version>3.3.1</maven3.3.x.version>
```
Comment (aheritier) [inline pom.xml]: I'm not sure if we should use the lowest or the highest version. For others versions it's nor the lowest or the highest version ... cc @olamy

**Comment by aheritier** [comment]
@tomerc could you squash your commits please ?

**Comment by tomerc** [comment]
I've squashed the commits

File: maven33-agent/src/main/java/jenkins/maven3/agent/Maven33Main.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    }
+	
+    /**
+     * Makes sure that this is Java5 or later.
+     */
+    private static void versionCheck() {
+        String v = System.getProperty("java.class.version");
+        if (v != null) {
+            try {
+                if (Float.parseFloat(v) < 49.0) {
```
Comment (olamy) [inline maven33-agent/src/main/java/jenkins/maven3/agent/Maven33Main.java]: I think this is wrong now as the requirement is different for 3.3.x

Comment (tomerc) [inline maven33-agent/src/main/java/jenkins/maven3/agent/Maven33Main.java]: @olamy You are correct of course, I've modified it for Java 7 class version (51)

**Comment by tomerc** [comment]
@aheritier @olamy Any news about this pull request? Thanks

**Comment by aheritier** [comment]
Hi @tomerc 

We discussed about it yesterday with @olamy 
As you saw the maven plugin was released and we will be able now to upgrade it to Java 7 and Jenkins > 1.612
I'm traveling next week thus I'll probably won't have the time to work on it but the week after I should be able to spend some time on it.

**Comment by tomerc** [comment]
@aheritier sounds great, thank you!

**Comment by aheritier** [comment]
Sorry for the delay @tomerc I discovered that I didn't have the required access here and it took some time to obtain it. I will move forward to update the module and release it. Then I'll integrate it in the maven plugin that I'll release

**Comment by tomerc** [comment]
@aheritier Thanks for all your help! Am eager to try out the new version of the plugin in a production environment.

**Pull request merged**